src/v/kafka/server/tests/BUILD

```diff
@@ -0,0 +1,42 @@
+load("//bazel:test.bzl", "redpanda_cc_btest")
+
+redpanda_cc_btest(
+    name = "atomic_token_bucket_test",
+    timeout = "short",
+    srcs = [
+        "atomic_token_bucket_test.cc",
+    ],
+    deps = [
+        "//src/v/kafka/server",
+        "//src/v/test_utils:seastar_boost",
+        "//src/v/utils:to_string",
+        "@boost//:test",
+    ],
+)
+
+redpanda_cc_btest(
+    name = "error_mapping_test",
+    timeout = "short",
+    srcs = [
+        "error_mapping_test.cc",
+    ],
+    deps = [
+        "//src/v/kafka/server",
+        "//src/v/test_utils:seastar_boost",
+        "@boost//:test",
+    ],
+)
+
+# TODO does this really not depend on kafka server?
```

Comment:
doesn't look like it. tests a single function from a single header in `kafka/protocol`, so perhaps it should live in there. out of scope for this diff, presumably?

Reply:
i think so. once this PR goes in then all of the kafka subsystem is covered by bazel. the enterprise team or anyone that feels compelled can definitely come back through and improve things. for example just looking at this PR, the server library can / should be broken down in more fine grained private libraries. a lot of that can probably be mechanical, but some of it may involve refactoring. it'd be great to get some of tyler's insight here.